apis/v1alpha1/xdp_program_types.go

```diff
@@ -22,18 +22,16 @@ type XdpProgramInfo struct {
 	// links is the list of points to which the program should be attached.  The list items
 	// are optional and may be udated after the bpf program has been loaded
 	// +optional
-	// +kubebuilder:default:={}
-	Links []XdpAttachInfo `json:"links"`
+	Links []XdpAttachInfo `json:"links,omitempty"`
 }
 
 type XdpAttachInfo struct {
 	// interfaceSelector to determine the network interface (or interfaces)
 	InterfaceSelector InterfaceSelector `json:"interfaceSelector"`
 
-	// containers identify the set of containers in which to attach the eBPF
-	// program.
-	// +optional
-	Containers ContainerSelector `json:"containers,omitempty"`
+	// networkNamespaces identifies the set of network namespaces in which to
+	// attach the eBPF program.
```

Comment:
why this isn't `optional` ?

Reply:
It's required for the network namespace-scoped programs.

Reply:
To expand further.  The ContainerSelector was optional on cluster objects, but required for namespace scoped objects.  I kept that the same when I replaced the ContainerSelector with a NetworkNamespaceSelector.